src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.Unix.cs

```diff
@@ -82,8 +82,10 @@ private TarEntry ConstructEntryForWriting(string fullPath, string entryName, Fil
             entry._header._gid = (int)status.Gid;
             if (!_groupIdentifiers.TryGetValue(status.Gid, out string? gName))
             {
-                gName = Interop.Sys.GetGroupName(status.Gid);
-                _groupIdentifiers.Add(status.Gid, gName);
+                if (Interop.Sys.TryGetGroupName(status.Gid, out gName) && gName != null)
```

Comment:
we know it's not null when the method returns true

```suggestion
                if (Interop.Sys.TryGetGroupName(status.Gid, out gName))
```

Reply:
[The violence worked](https://github.com/dotnet/runtime/pull/81070/files#r1085707454).